### PR TITLE
feat: create frameless buttons

### PR DESCRIPTION
**Pull request by luqven**
## Video 📹 

https://user-images.githubusercontent.com/16711614/150616584-2d96b7a9-7c63-4e84-9bfa-4486001e8226.mov

https://user-images.githubusercontent.com/16711614/150616590-8db6c36a-3d02-463d-8c18-3e997a204a1a.mov

### CODE REVIEW

**Comment by luqven** [comment]
Current dependencies on/for this PR:
* next
  * **PR #184** 
    * **PR #185** 
      * **PR #193** 
        * **PR #192** 
          * **PR #197** 
            * **PR #194** 
              * **PR #191** 
                * **PR #201** 
                  * **PR #196**  (this PR)
                    * **PR #198** 
                      * **PR #199** 
                        * **PR #200** 
                          * **PR #195** 
      * **PR #187** 
        * **PR #186** 

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).

File: frontend/src/components/buttons/FrameButton/FrameButton.module.scss [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  }
+  &.iconOnly {
+    min-width: 36px;
+    min-height: 36px;
+  }
+
+  &.large {
+    height: 36px;
+  }
+  // storybook is bugging out and wont' let me remove the medium option
```
Comment (frederickfogerty) [inline frontend/src/components/buttons/FrameButton/FrameButton.module.scss:29]:
```suggestion
  // storybook is bugging out and won't let me remove the medium option
```

**Review by frederickfogerty — CHANGES_REQUESTED** [review]
I like the implementation here but I'm wondering about the purpose - we already have a `Button` component - can we work the functionality here into that component potentially?

**Comment by luqven** [comment]
> I like the implementation here but I'm wondering about the purpose - we already have a `Button` component - can we work the functionality here into that component potentially?

Yes, we do already have a `Button` module. The issue I was running into with it was frustrating enough though that I wanted to start from scratch:

- controlling the colors/size was more difficult and often different from the design guide examples
- controlling for behavior when there was no text/icon was more difficult and often the end result was not centered correctly
- Drop shadows, frames, etc could not be turned on/off easily depending on props, which meant I needed to use wrappers often.

The one functionality I'd like to bring over from the existing module is the `flat` prop. Otherwise, I think we'll be better served by this module moving forward.

The only outstanding concern I have with this module (which I still have to test) is whether or not we need to keep using `div` instead of `button`, as I'm doing where. I'm a fan of the semantic web but I do remember at one point we had an issue where using `button` caused some unnecessary `submit` event firing.

If you'd rather I backfill these changes into the existing button we can definitely do that instead. Let me know what you think.

**Comment by luqven** [comment]
I'm going to merge in this module for now to unblock myself but definitely open to reverting to the existing module/porting this module features over.

**Comment by frederickfogerty** [comment]
I understand your pain around the existing Button component, but without taking the effort to migrate the existing usages of the old Button component to this one, then we end up in a worse situation. I think we should definitely take time to ensure that we are just using one component rather than two.

**Review by frederickfogerty — COMMENTED** [review]
See PR comment

**Pull request merged**